test/simulator/asm/org/apache/cassandra/simulator/asm/ClassTransformer.java

```diff
@@ -137,12 +139,58 @@ private ClassTransformer(int api, ClassWriter classWriter, String className, Enu
         this.methodLogger = MethodLogger.log(api, className);
     }
 
+    public void setUpdateVisibility(boolean updateVisibility)
+    {
+        this.updateVisibility = updateVisibility;
+    }
+
+    /**
+     * Java 11 changed the way that classes defined in the same source file get access to private state (see https://openjdk.org/jeps/181),
+     * rather than trying to adapt to this, this method attempts to make the field/method/class public so that access
+     * is not restricted.
+     */
+    private int makePublic(int access)
```

Comment:
This will be only for the simulator ?

Reply:
Yep, this is in the simulator tree so I don't believe it's even available at compilation or runtime to other parts of the codebase.

Reply:
yes, this is part of the simulator; we rewrite classes so we can control execution order... but on JDK 11 how private fields/methods are shard with subclasses changed, so rather than trying to make things complex, the rewrite logic makes everything public

Reply:
Yup, that's what I thought, thanks. For testing and being only used in the simulator sounds reasonable to me